test/models/sync_plan_test.rb

```diff
@@ -129,5 +129,32 @@ def test_schedule_format_disabled
       assert_match(/\/PT24H$/, schedule)
       assert_includes schedule, @plan.sync_date.iso8601
     end
+
+    def test_add_product
+      product = katello_products(:redhat)
+      @plan.products << product
+      @plan.save!
+      assert_equal product, @plan.products.first
+      assert_equal 1, @plan.products.count
+    end
+
+    def test_remove_product
+      product = katello_products(:redhat)
+      @plan.products << product
+      @plan.save!
+      assert_equal product, @plan.products.first
+      assert_equal 1, @plan.products.count
+      @plan.products.clear
+      @plan.save!
+      assert_empty @plan.products
+    end
+
+    def test_empty_add_product
+      product = katello_products(:empty_redhat)
+      @plan.products << product
+      assert_raises(ActiveRecord::RecordInvalid) do
```

Comment:
You can make this test more efficient by just testing the validation with something like this:

```ruby
refute(@plan.valid?, "Plan must be invalid")
```